common/web/input-processor/src/text/prediction/languageProcessor.ts

```diff
@@ -127,8 +127,10 @@ export default class LanguageProcessor extends EventEmitter<LanguageProcessorEve
 
     return this.lmEngine.loadModel(source, specType).then((config: Configuration) => {
       this.configuration = config;
-      this._state = 'configured';
-      this.emit('statechange', 'configured');
+      if(this.mayPredict) {
+        this._state = 'configured';
+        this.emit('statechange', 'configured');
+      }
```

Comment:
> I don't really understand how change to languageProcessor.ts addresses the issue. Can you elaborate?

From the linked comment that describes the issue:

> 1. The lexical model has changed from the one most recently active - thus, we deactivate predictions momentarily.
> 2. Web starts loading the new lexical model.
> 3. A Web event signals this change in a way that Android picks up.
> 4. Android sees "oh, no predictions are active [in the Web engine] - disable the prediction banner.
> 5. Web finishes loading the new lexical model and signals this - predictions available!
>     - Bug:  despite predictions being set to off!
> 6. Android sees "predictions ready!" - enable the prediction banner.
> 7. Web is told "enable predictions" - which _retriggers_ the "predictions not ready" event due to the following two bits of logic:
>     https://github.com/keymanapp/keyman/blob/5c6fbcc418a1709f278c0d185c8eaac644cc305a/android/KMEA/app/src/main/assets/android-host.js#L152-L154
> 
>     (i.m.o, the 'configured'-only conditional should likely be considered a bug.  Less critical, but it obviously contributed to this!)
> 
>     https://github.com/keymanapp/keyman/blob/5c6fbcc418a1709f278c0d185c8eaac644cc305a/common/web/input-processor/src/text/prediction/languageProcessor.ts#L416-L429
> 8. Noting the latter block above, Web essentially queues two events:  "predictions not ready" and "predictions ready".
> 9. Android thus sees "predictions not ready" and says "deactivate predictions".
> 10.  And then sees "predictions ready" and says "activate predictions".
> 11. Web receives the first one first... retriggering step 7 above, causing an infinite event loop!

Note step 5 above and the 'bug' note below it.  That's exactly what the changes here fix. 

Since the lexical model loads asynchronously, it is possible to have `mayPredict == 'true'` when the `loadModel` call occurs but to have `mayPredict == false` afterward, within the affected codeblock.  We even bother to check in lines 123 -126 immediately before the `loadModel` call and prevent one emission there; therefore, we should do the same here too.